(PDB-3077) Add service reload and logback size-based rotation

camlow325 opened the pull request:
This PR bumps the Trapperkeeper and lein-ezbake dependencies up to 1.5.1 and 1.0.0, respectively, in order for PuppetDB packages to be built with support for service reload functionality (built on top of SIGHUP).  This PR also includes some logback configuration changes to support moving to logback file-size based log rotation and disabling the logic in ezbake that previously enabled log rotation via logrotate.

camlow325 commented:
I noticed that there are some helpers in the beaker tests that do full stops / starts of the puppetdb service that could be moved over to using `service puppetdb reload` instead, such as ![restart_puppetdb](https://github.com/puppetlabs/puppetdb/blob/ac03a89079883d71297252cc3f5530b3e45e3c4d/acceptance/helper.rb#L504-L507).  I didn't bother trying to change those in this PR as I wasn't sure if the PuppetDB team thought we might be losing valuable existing test coverage in the process.  Moving some of these to `reload`, though, could provide some performance improvements so I figured I'd point that out as potential future work.

camlow325 commented:
Ping @cprice404 and @jpinsonault for additional review.

camlow325 commented:
In addition to running the Clojure unit tests locally to check for any regressions related to the TK dependency bump, I installed a PuppetDB package built with these changes and manually smoke tested the use of the service `reload` and logback rotation on the puppetdb.log and puppetdb-access.log files.  All seemed to work as I'd expected it to.

puppetlabs-jenkins commented:
Test PASSed

puppetcla commented:
CLA signed by all contributors.

cprice404 commented on resources/ext/config/logback.xml:
```diff
@@ -7,10 +7,13 @@
+            <!-- each file should be at most 10MB, keep 90 days worth of history, but at most 1GB total-->
+            <maxFileSize>10MB</maxFileSize>
             <maxHistory>90</maxHistory>
+            <totalSizeCap>1GB</totalSizeCap>
```
Were we going to go ahead and add the snippet for the status debug logging in here while we were messing with this file?  TK-400?

cprice404 commented:
LGTM

camlow325 commented:
@cprice404 I added the status log debugging snippet in 2701971.  Hope you and @jpinsonault might take a peek to see that it looks like what you were expecting.  I'll at least do some smoke testing with this to see that it doesn't break existing logging.

jpinsonault commented:
A puppetdb-status.log file is created even if nothing has tried to log to it?

camlow325 commented:
> A puppetdb-status.log file is created even if nothing has tried to log to it?

Yeah, apparently.  That's what I saw with the logback config on this PR anyway.

jpinsonault commented:
Oh that's unfortunate, but not terrible I guess

jpinsonault commented:
I'm 👍

puppetlabs-jenkins commented:
Test PASSed

camlow325 commented:
Rebased this at 26c2a5d.  In the process, I fixed a dumb mistake I'd made where I used .gz in one of the logback xml files as the rotated log file instead of .zip.  .zip is the format that logback uses for its log rotation.

puppetlabs-jenkins commented:
Test PASSed

camlow325 commented:
After talking about this further with @cprice404 and @jpinsonault, we decided it would be better to use .gz instead of .zip for the compressed log archives.  That's what PuppetDB was already using anyway - which I'd initially overlooked.  I amended the comments that added the logging segments to switch out the extension.  I'll do a sanity check with a local ezbake build to see that this all seems to be working properly.

puppetlabs-jenkins commented:
Test PASSed

camlow325 commented:
In 4a95495, I changed the `maxFileSize` for the different log files from 10 MB to 200 MB, based on an offline conversation with @wkalt and @cprice404.

puppetlabs-jenkins commented:
Test PASSed

camlow325 commented:
@wkalt - are you +1 to merge this PR at this point?  Let me know if you think anything else needs to change first.  Thanks!
